analyses/data-pre-release-qc/01-data-harmonization-qc.Rmd

```diff
@@ -0,0 +1,752 @@
+  output_file <- "methyl-m-values-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("Methylation biospecimen in methyl-m-values.rds missing in histolgies = ", length(missing_hist_ids)))
+```
+
+- CN-values matrix (methyl-m-values.rds)
+```{r}
+# check that all methylation biospecimen IDs are in the m-values methylation matrix
+biospecimen_ids <- readr::read_rds(file.path(data_dir, "methyl-cn-values.rds")) %>%
+  dplyr::select(-Probe_ID) %>%
+  colnames()
+missing_ids <- setdiff(hist_ids, biospecimen_ids)
+missing_hist_ids <- setdiff(biospecimen_ids, hist_ids)
+if (length(missing_ids) > 0) {
+  output_file <- "histologies-samples-missing-in-methyl-cn-values.tsv"
+  missing_ids_df <- missing_biospecimen_table(hist_df, missing_ids, output_file)
+  missing_ids_df
+}
+print(paste("Methylation biospecimen in histolgies missing in methyl-cn-values.rds = ", length(missing_ids)))
+if (length(missing_hist_ids) > 0) {
+  output_file <- "methyl-cn-values-samples-missing-in-histologies.tsv"
+  missing_hist_ids_df <- tibble::tibble(Kids_First_Biospecimen_ID = missing_hist_ids)
+  missing_hist_ids_df %>% readr::write_tsv(file.path(results_dir, output_file))
+  missing_hist_ids_df
+}
+print(paste("Methylation biospecimen in methyl-cn-values.rds missing in histolgies = ", length(missing_hist_ids)))
+```
+
+#### Check methylation matrices probe IDs
+
+- Methylation matrices
+```{r}
+# beta-values matrix
+beta_values <-
+  tibble::tibble(beta_values =
+    readr::read_rds(file.path(data_dir, "methyl-beta-values.rds")) %>%
+      dplyr::pull(Probe_ID) %>% unique()) %>%
+  dplyr::mutate(probe_id = beta_values)
+
+# m-values matrix
+m_values <-
+  tibble::tibble(m_values =
+    readr::read_rds(file.path(data_dir, "methyl-m-values.rds")) %>%
+      dplyr::pull(Probe_ID) %>% unique()) %>%
+  dplyr::mutate(probe_id = m_values)
+
+# cn-values matrix
+cn_values <-
+  tibble::tibble(cn_values =
+    readr::read_rds(file.path(data_dir, "methyl-cn-values.rds")) %>%
+      dplyr::pull(Probe_ID) %>% unique()) %>%
+  dplyr::mutate(probe_id = cn_values)
+
+# merge and list differences
+probe_id_diffs <- beta_values %>%
+  dplyr::full_join(m_values, by = "probe_id") %>%
+  dplyr::full_join(cn_values, by = "probe_id") %>%
+  dplyr::select(-probe_id) %>%
+  dplyr::filter(is.na(beta_values) | is.na(m_values) | is.na(cn_values)) %>%
+  dplyr::distinct()
+
+# check any differences in probe IDs among methyl matrices
+if (length(rownames(probe_id_diffs)) > 0) {
+  output_file <- "probes_not_in_all_methyl_matrices.tsv"
+  gene_symbol_diffs %>% readr::write_tsv(file.path(results_dir, output_file))
+  gene_symbol_diffs
+}
+print(paste("Probes not matching all methylation matrices =", length(rownames(probe_id_diffs))))
+```
+
+#### Check DNA SNV, CNV and SV biospecimen IDs
```

Comment:
Only SNV will likely have all WXS+WGS samples from histology file in the MAF. Targeted Panel files may very well be missing if no calls. 

No CNV was called on panel data.